### PR TITLE
add elastic inference sagemaker integ test

### PR DESCRIPTION
**Pull request by ChoiByungWook**
*Issue #, if available:*

*Description of changes:*
Added EI integ test

Test results:

**=== non-supported region ===**
pytest -rs test/integration/sagemaker/test_elastic_inference.py --accelerator-type ml.eia1.medium --region ap-southeast-1

test/integration/sagemaker/test_elastic_inference.py::test_deploy_elastic_inference_with_pretrained_model SKIPPED
============================================================================== short test summary info ==============================================================================
SKIP [1] /Users/choidan/workspace/container/sagemaker-mxnet-containers/test/integration/sagemaker/test_elastic_inference.py:45: EI is not supported in ap-southeast-1

**=== no accelerator provided ===**
pytest test/integration/sagemaker/test_elastic_inference.py

test/integration/sagemaker/test_elastic_inference.py::test_deploy_elastic_inference_with_pretrained_model SKIPPED
============================================================================== short test summary info ==============================================================================
/Users/choidan/workspace/container/sagemaker-mxnet-containers/test/integration/sagemaker/test_elastic_inference.py:39: Skipping because accelerator type was not provided

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integration/__init__.py [hunk 1]
```diff
@@ -24,3 +24,7 @@
 # Workaround for the intermittent worker timeout errors
 # TODO: find and solve the root cause of this issue
 NUM_MODEL_SERVER_WORKERS = 2
+
+# EI is currently only supported in the following regions
+# regions were derived from https://aws.amazon.com/machine-learning/elastic-inference/pricing/
+EI_SUPPORTED_REGIONS = ['us-east-1', 'us-east-2', 'us-west-2', 'eu-west-1', 'ap-northeast-1', 'ap-northeast-2']
```
Comment (laurenyu) [inline test/integration/__init__.py]: can we read this from the Python SDK or something? just getting wary of having this hardcoded in too many places.

File: test/integration/sagemaker/test_elastic_inference.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+@pytest.fixture(autouse=True)
+def skip_if_non_supported_ei_region(region):
+    if region not in EI_SUPPORTED_REGIONS:
+        pytest.skip('EI is not supported in {}'.format(region))
+
+
+@pytest.fixture
+def pretrained_model_data(region):
+    return 's3://sagemaker-sample-data-{}/mxnet/model/resnet/resnet_50.tar.gz'.format(region)
```
Comment (laurenyu) [inline test/integration/sagemaker/test_elastic_inference.py]: just want to make sure - has this been uploaded in all the expected regions already?

File: test/integration/__init__.py (hunk 1 above)
Comment (ChoiByungWook) [reply test/integration/__init__.py]: It is in the init file in the test directory. I can't pull it out, however you do make a good point still.

Comment (laurenyu) [reply test/integration/__init__.py]: fair enough

File: test/integration/sagemaker/test_elastic_inference.py (hunk 2 above)
Comment (ChoiByungWook) [reply test/integration/sagemaker/test_elastic_inference.py]: aws s3 ls s3://sagemaker-sample-data-us-west-1/mxnet/model/resnet/resnet_50.tar.gz
2019-02-15 15:05:25  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-us-east-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:09:56  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-us-east-2/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:10:12  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-eu-west-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:10:26  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-eu-west-2/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:10:57  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-eu-central-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:11:10  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-ca-central-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:11:29  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-ap-northeast-2/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:11:40  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-ap-northeast-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:11:56  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-ap-south-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:12:23  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-ap-southeast-1/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:13:15  225557606 resnet_50.tar.gz
aws s3 ls s3://sagemaker-sample-data-ap-southeast-2/mxnet/model/resnet/resnet_50.tar.gz
2018-11-15 15:13:42  225557606 resnet_50.tar.gz

File: README.rst [hunk 3]
```diff
@@ -309,6 +309,19 @@ To run SageMaker integration tests:
                                       --instance-type ml.m4.xlarge \
                                       --tag 1.3.0-cpu-py3
 
+If you want to run a SageMaker end to end test for your Elastic Inference container, you will need to provide an `accelerator_type` as an additional pytest argument.
+
+The `accelerator-type` is your specified `Amazon Elastic Inference Accelerator <https://aws.amazon.com/sagemaker/pricing/instance-types/>`__ type that will be attached to your instance type.
+
+::
+
+    # Example for running Elastic Inference functional test
```
Comment (laurenyu) [inline README.rst]: don't call it a functional test

File: test/integration/sagemaker/test_elastic_inference.py (hunk 2 above)
Comment (ChoiByungWook) [reply test/integration/sagemaker/test_elastic_inference.py]: it isn't in pdt

**Pull request merged**